xarray/backends/h5netcdf_.py

```diff
@@ -431,11 +430,72 @@ def open_dataset(  # type: ignore[override]  # allow LSP violation, not supporti
     def open_datatree(
         self,
         filename_or_obj: str | os.PathLike[Any] | BufferedIOBase | AbstractDataStore,
+        mask_and_scale=True,
```

Comment:
```suggestion
	*,
        mask_and_scale=True,
```
separate positional from keyword args like you do in the netcdf file.